dsl/src/state_generator.rs

```diff
@@ -38,8 +38,20 @@ fn render_state(s: &State) -> String {
     State::StorageCreateItemRequest { storage } => format!("{}StorageCreateItemRequest", storage),
 
     State::FuncEntry { func } => format!("{}Entry", func),
-    State::FuncRecursiveCall { func } => format!("{}RecursiveCall", func),
-    State::FuncCall { func, callee } => format!("{}Call{}", func, callee),
+    State::FuncRecursiveCall { func, index } => {
+      if *index == 0 {
```

Comment:
Perhaps start with `1` and continue from `2`? If you want to skip `1` that looks good, but `Call` and `Call2` makes more sense to me than `Call` and `Call1`.